keystore/src/transaction/mod.rs

```diff
@@ -283,30 +282,58 @@ impl KeystoreTransaction {
-                (identifier_10, MlsPendingMessage),
-                (identifier_11, E2eiEnrollment),
-                (identifier_12, E2eiRefreshToken),
-                (identifier_13, E2eiAcmeCA),
-                (identifier_14, E2eiIntermediateCert),
-                (identifier_15, E2eiCrl),
-                (identifier_16, ProteusPrekey),
-                (identifier_17, ProteusIdentity),
-                (identifier_18, ProteusSession)
-            ]
-        );
+        cfg_if::cfg_if! {
+            if #[cfg(feature = "proteus-keystore")] {
+                commit_transaction!(
+                    self, db,
+                    [
+                        (identifier_01, MlsCredential),
+                        (identifier_02, MlsSignatureKeyPair),
+                        (identifier_03, MlsHpkePrivateKey),
+                        (identifier_04, MlsEncryptionKeyPair),
+                        (identifier_05, MlsEpochEncryptionKeyPair),
+                        (identifier_06, MlsPskBundle),
+                        (identifier_07, MlsKeyPackage),
+                        (identifier_08, PersistedMlsGroup),
+                        (identifier_09, PersistedMlsPendingGroup),
+                        (identifier_10, MlsPendingMessage),
+                        (identifier_11, E2eiEnrollment),
+                        (identifier_12, E2eiRefreshToken),
+                        (identifier_13, E2eiAcmeCA),
+                        (identifier_14, E2eiIntermediateCert),
+                        (identifier_15, E2eiCrl),
+                        (identifier_16, ProteusPrekey),
+                        (identifier_17, ProteusIdentity),
+                        (identifier_18, ProteusSession)
+                    ]
+                );
+            }
+        }
 
+        cfg_if::cfg_if! {
+            if #[cfg(not(feature = "proteus-keystore"))] {
+                commit_transaction!(
+                    self, db,
+                    [
+                        (identifier_01, MlsCredential),
+                        (identifier_02, MlsSignatureKeyPair),
+                        (identifier_03, MlsHpkePrivateKey),
+                        (identifier_04, MlsEncryptionKeyPair),
+                        (identifier_05, MlsEpochEncryptionKeyPair),
+                        (identifier_06, MlsPskBundle),
+                        (identifier_07, MlsKeyPackage),
+                        (identifier_08, PersistedMlsGroup),
+                        (identifier_09, PersistedMlsPendingGroup),
+                        (identifier_10, MlsPendingMessage),
+                        (identifier_11, E2eiEnrollment),
+                        (identifier_12, E2eiRefreshToken),
+                        (identifier_13, E2eiAcmeCA),
+                        (identifier_14, E2eiIntermediateCert),
+                        (identifier_15, E2eiCrl)
+                    ]
+                );
+            }
+        }
```

Comment:
Can we make this more concise by moving the Proteus-specific parts under their own `cfg_if` block?